Disappearing Messages V2

yougotwill opened the pull request:
- Updated right panel in a conversation to use overlay logic similar to the left panel.
- Improved design for disappearing messages
- Added panel radio buttons that are used for the disappearing message timer options
- Improved Conversation Header that can switch between the notification settings, disappearing message setting and group members. When you click on the heading it opens the related right panel option (if available)
- Added 2 new disappearing message modes (The original mode aka. Legacy will be removed in the future but will be kept for backwards compatibility / migration purposes for now)
  - Disappear After Read
  - Disappear After Send
- Added disappearing support to other visible messages (call, group update, media extraction notifications etc.)
- Use clock svg for disappearing countdown in all cases to keep in line with other platforms.
- Disappearing messages are now correctly deleted from the swarm and won't appear when restoring a device
- Users with newer desktop clients will get notified if they are sent messages from an older (legacy) client encouraging them to tell the other user to upgrade
- Attachments should be correctly deleted off disk (and off swarm) once deleted in the client (solves
https://github.com/oxen-io/session-desktop-temp/issues/315, https://github.com/oxen-io/session-desktop/issues/
- Improved spacing and line height on toast notification
- Disappearing Messages settings now sync via libsession with release [v0.3.0](https://github.com/oxen-io/libsession-util-nodejs/pull/9)
- Fixed visual alignment all message types in message list and moved spacing to parent. This also fixes the alignment of hte disappearing messages timer for different message types
- Unit Tests everywhere

yougotwill committed:
fix: cleanup

yougotwill committed:
fix: make sure disappearing messages is the first item in the conversation header subtitle

yougotwill committed:
fix: make sure lastDisappearingMessageChangeTimestamp is defined before comparing with convo model

yougotwill committed:
feat: tested v2 releas

made some updates to expire update generation and handling, one final fix required

yougotwill committed:
fix: show disappearing messages subtitle in convo header

when it is toggled on

yougotwill committed:
fix: whitespace in comment

yougotwill committed:
feat: added temp set feature for testing

yougotwill committed:
feat: use checkForExpireUpdateInContentMessage on sync messages

yougotwill committed:
feat: shouldDisappearButIsntMessage is now a function

result is added to the expireUpdate and improves outdated client checking logic

yougotwill committed:
feat: outdated banner now shows specific messaging for synced messages

yougotwill committed:
fix: deleteAfterSend now works in private convos again

yougotwill committed:
fix: single mode should show the correct timer values

if a mode is invalid we return no timer options

yougotwill committed:
fix: improved logging

made some progress with fixing things but other things broke

yougotwill committed:
fix: newlines for logging

yougotwill committed:
fix: single mode value should be off

if disappearing messages isnt on

yougotwill committed:
fix: ignore equal lastMessageChangeTimestamps

yougotwill committed:
fix: don't send sync messages for outdated changes

yougotwill committed:
fix: added isOutdated prop to expireUpdate

ignores duplicate setting messages

yougotwill committed:
fix: make sure to note sent another message

in all remote cases

yougotwill committed:
fix: outgoing content messages can have an equal changetimestamp

yougotwill committed:
fix: log pipeline better

yougotwill committed:
fix: canBeDeleteAftereSend in messagejob was wrong

yougotwill committed:
fix: regular messages correctly sync again

yougotwill committed:
fix: trigger expiry for outgoing sync messages

yougotwill committed:
fix: sync messages now expire in 1on1s

fromConfig has been removed from updateExpireTimer

yougotwill committed:
Merge branch 'unstable' into userconfig_disappearingmessage

yougotwill committed:
fix: group invitation timer alignment

updated data extraction notification design. Just waiting on icon from ios

yougotwill committed:
feat: added new save icon

updated DataExtractNotification and Lightbox download icons

yougotwill committed:
fix: use updateExpireTimer for UserProfile

before we were setting the convo values directly

yougotwill committed:
fix: show durations for constants on hover

for TTL_DEFAULT and SWARM_POLLING_TIMEOUT

yougotwill committed:
fix: use TTL_CONFIG in retrieve request

also some comment cleanup

yougotwill committed:
feat: use SubRequest typings for expireOnNodes

created request builder, added logging for now to test

yougotwill committed:
feat: adding proper typings to snode expire code

I dont think the expire value returned is correct, needs debugging

yougotwill committed:
feat: added retry logic to expireMessageOnSnode

yougotwill committed:
feat: make sure to use ms for the expireTimer

expireTimer on the model is in seconds so we need to convert it

yougotwill committed:
fix: comment out some unnecessary logging

yougotwill committed:
feat: update the TTL with the snode expire result

yougotwill committed:
feat: started get_expiries work

signature now generates and renamed expire file to expireRequest

yougotwill committed:
feat: added get_expiries typings

yougotwill committed:
feat: improved logging for start expiry timestamp

yougotwill committed:
feat: improved expire swarm result logic

better logging and handle errors more smoothly

yougotwill committed:
fix: remove redundant start timestamp start in read receipts

yougotwill committed:
feat: get_expiries implementation done

now just need to make sure to call both endpoints in the correct places

yougotwill committed:
fix: we can trust the read start from libsession for expirationTimerUpdates

therefore we don't need to update the read status explicitly

yougotwill committed:
fix: hit expire endpoint when send is successful

we need the returned messageHash from the swarm

yougotwill committed:
feat: hit expire endpoint for read messages

for now it will hit the expire endpoint twice

yougotwill committed:
fix: commented out some logs

yougotwill committed:
fix: add temp fix for single msg get_expiries

yougotwill committed:
feat: improved expiry retry and error handing

nested pRetry Abort errors can break our retry logic so we need to catch them

yougotwill committed:
fix: get_expiries retry and error handing

nested pRetry Abort errors can break our retry logic so we need to catch them

yougotwill committed:
refactor: disappearingmessageconversationtype to disappearingmessageconversationmodetype

yougotwill committed:
refactor: rename expirationType on convo to expirationMode

this means convo has expirationMode and messages have expirationType

yougotwill committed:
fix: messages are no longer randomly deleted

forgot to expire only unread disappearing messages and not all messages

yougotwill committed:
refactor: changetodisappearingmessageconversationtype to changetodisappearingconversationmode

yougotwill committed:
refactor: useselectedconversationexpirationtype to useselectedconversationdisappearingmode

yougotwill committed:
refactor: providedexpirationtype to provideddisappearingmode

yougotwill committed:
fix: single mode is off by default

yougotwill committed:
feat:  anyone in a group can change the setting

before the v2 release

yougotwill committed:
feat: dont expire messages that fail to send

yougotwill committed:
feat: when v2 is released migration legacy convos

use the default mode and timer value for the convo type

yougotwill committed:
fix: conversationheadertitle shortens pubkey

yougotwill committed:
refactor: rename hook

useconversationusernameorshorten to usenicknameorprofilenameorshortenedpubkey

yougotwill committed:
feat: created getters on convo model

replaced all expirationMode calls

yougotwill committed:
refactor: replaced get expireTimer calls

with getExpireTimer

yougotwill committed:
refactor: replaced get hasOutdatedClient calls

with getHasOudatedClient

yougotwill committed:
refactor: replaced get lastDisappearingMessageChangeTimestamp calls

with getLastDisappearingMessageChangeTimestamp

yougotwill committed:
feat: created getters in message model

replaced all expirationType calls with getExpirationType

yougotwill committed:
refactor: replaced get expireTimer calls

with getExpireTimer

yougotwill committed:
refactor: replaced get expirationStartTimestamp calls

with getExpirationStartTimestamp

yougotwill committed:
refactor: replaced get expires_at calls

with getExpiresAt

yougotwill committed:
refactor: replaced get expirationTimerUpdate calls

with getExpirationTimerUpdate

yougotwill committed:
fix: sync legacy mode in single mode before V2

for NTS and closed groups we want to set the expirationMode to legacy

yougotwill committed:
fix: disappear synced settings updates

need to trigger the expirationStartTimestamp manually

yougotwill committed:
feat: updateExpireTimer now returns success

we use this to decide on commiting updates from sync messages to the db

yougotwill committed:
feat: commit disappearing remote changes to db

yougotwill committed:
Merge branch 'unstable' into userconfig_disappearingmessage

yougotwill committed:
fix: correct getter region comments

yougotwill committed:
fix: dont load container if there are no reactions

now message elements are the correct height from what we see visually

yougotwill committed:
fix: remove alignment from ExpirableReadableMessage

improved alignment of timer by moving into the MessageContentWithStatus

yougotwill committed:
fix: timer position for GenericReadableMessages

still need to do other messages

yougotwill committed:
fix: timer alignement for group invitation

yougotwill committed:
fix: timer alignment for call notifications

yougotwill committed:
fix: timer alignment for timer notifications

yougotwill committed:
fix: timer alignment for data extraction notifications

yougotwill committed:
fix: timer alignment for group update

yougotwill committed:
feat: move padding globally to the messages list

we dont want individual padding on each message type

yougotwill committed:
fix: removed public-chat-message-wrapper padding

Bilb commented on ts/components/conversation/header/ConversationHeaderTitle.tsx:
```diff
@@ -0,0 +1,198 @@
+      : null;
+  }, [expirationMode, expireTimer, isGroup, isMe, isPublic]);
+
+  const handleRightPanelToggle = () => {
+    if (isRightPanelOn) {
+      dispatch(closeRightPanel());
+    } else {
+      if (visibleSubtitle === 'disappearingMessages') {
+        dispatch(setRightOverlayMode('disappearing-messages'));
+      } else {
+        dispatch(resetRightOverlayMode());
+      }
+      dispatch(openRightPanel());
+    }
+  };
```
early return and no else?

Bilb commented on ts/components/conversation/message/message-item/ExpirableReadableMessage.tsx:
```diff
@@ -0,0 +1,153 @@
+        />
+      )}
+      {props.children}
+      {expirationLength && expirationTimestamp && (
```
ternary here? Or make it a boolean explicitely

Bilb commented on ts/components/conversation/right-panel/overlay/OverlayRightPanelSettings.tsx:
```diff
@@ -334,27 +341,33 @@ export const SessionRightPanelWithDetails = () => {
 
       <MediaGallery documents={documents} media={media} />
       {isGroup && (
+        // tslint:disable-next-line: use-simple-attributes
```
no more tslint right?

Bilb replied:
no more tslint right?

yougotwill committed:
fix: align expire timers with the member avatars

in groups

yougotwill committed:
fix: messages are now the right way up

I have broken the reverse flexbox

yougotwill committed:
fix: messages are now vertically spaced correctly

yougotwill committed:
feat: backported user_profile test

yougotwill committed:
feat: setup test outline for disappearing messages

yougotwill committed:
feat: added isValidUnixTimestamp

yougotwill committed:
test: start setExpirationStartTimestamp

tested different modes

yougotwill committed:
test: finished setExpirationStartTimestamp tests

yougotwill committed:
test: finished changeToDisappearingMessageType tests

yougotwill committed:
test: finished changeToDisappearingMessageType tests

the last commit was actually for changeToDisappearingConversationMode

yougotwill committed:
fix: remove unneeded comment

yougotwill committed:
feat: made ExpirationTimerUpdate type

yougotwill committed:
fix: resolved lint errors for hasDefinedProperty

part of the protobuf utils

yougotwill committed:
test: function for generating disappearing messages

yougotwill committed:
test: finished checkForExpireUpdateInContentMessage

yougotwill committed:
feat: updateExpireTimer arg shouldCommitMessage

when testing we dont want to commit the timer update message

yougotwill committed:
test: started updateExpireTimer

yougotwill committed:
refactor: rename generateDisappearingMessage

to generateDisappearingVisibleMessage

yougotwill committed:
test: finished isExpirationTimerUpdate

added generateFakeExpirationTimerUpdate for testing

yougotwill committed:
test: added false case for isExpirationTimerUpdate

yougotwill committed:
test: started expireRequest

added tests for buildExpireRequest

yougotwill committed:
test: finished verifyExpireMsgsResponseSignature

yougotwill committed:
test: started processExpireRequestResponse

yougotwill committed:
test: finished processExpireRequestResponse

yougotwill committed:
test: wip getExpiriesRequest Started

yougotwill committed:
test: finished buildGetExpiriesRequest

added TypedStub type

yougotwill committed:
test: finished processGetExpiriesRequestResponse

yougotwill committed:
refactor: remove duplicate test

libsession_wrapper_test.ts

yougotwill committed:
chore: remove unneeded stubWindowLog calls

yougotwill committed:
test: throw when we want a defined result

this fixings the typings with test when look for props on objects

yougotwill committed:
test: started insertContactFromDBIntoWrapperAndRefresh

yougotwill committed:
fix: dont use ourNumber in contacts tests

KeeJef committed:
docs: update I18n section

yougotwill committed:
test: finished libsession_wrapper_user_profile tests

yougotwill committed:
fix: use generated pubkeys

in disappearing message tests

yougotwill committed:
test: stub libsession workers in all places

instead of stub the set function

yougotwill committed:
test: started libsession LegacyGroups tests

specifically insertGroupsFromDBIntoWrapperAndRefresh

yougotwill committed:
refactor: fixed comment typo

returned for testing purposes only

yougotwill committed:
test: fix groups wrapper values test

yougotwill committed:
test: finished insertGroupsFromDBIntoWrapperAndRefresh

yougotwill committed:
fix: make sure to use convo getters

instead of backbone get function

yougotwill committed:
test: updated ChatMessage with disappearing tests

for both modes

yougotwill committed:
fix: unblockGroupToSend copy change

yougotwill committed:
fix: remove ClosedGroupControlMessage comments

yougotwill committed:
fix: update legacy comment in signal service

this makes it easier for cleanup later

yougotwill committed:
fix: removed noBackgroundColor from panel buttons

yougotwill committed:
fix: remove unneeded tslint comment

yougotwill committed:
fix: cleanup NoticeBanner jsx

yougotwill committed:
fix: remove options menu from convo header

in favour of right panel

yougotwill committed:
fix: styledDot key should be more specific

yougotwill committed:
feat: use custom hook useIsRightPanelShowing

in functional components

yougotwill committed:
fix: remove else and return early

in handleRightPanelToggle

yougotwill committed:
fix: use convoId instead of convoName in dep array

for what subtitle to show when changing convos

yougotwill committed:
fix: expiretimer now uses a ternary operator

yougotwill committed:
fix: type state correctly when using useSelector

yougotwill committed:
feat: made datatestids mandatory on readablemessages

yougotwill committed:
feat: use custom hook useRightOverlayMode

yougotwill committed:
fix: remove all unneeded tslint comments

fixed some eslint errors

yougotwill committed:
fix: use precasted variables for modes

for DisappearingMessageConversationModeType references

yougotwill committed:
refactor: getDisappearingUnreadByConversation

to getUnreadDisappearingByConversation since reading causes disappearing

yougotwill committed:
feat: move timer options out of redux

yougotwill committed:
feat: moved disappearing logic to a feature folder

only moved types and timerOptions so far

yougotwill committed:
feat: moved remaining code into feature folder

also separated out legacy functions

yougotwill committed:
test: restructed tests to match feature folder

yougotwill committed:
fix: replace backbone get method

for lastDisappearingMessageChangeTimestamp

yougotwill committed:
fix: remove unnecessary async on markReadNoCommit

yougotwill committed:
refactor: isDisappearing to matchesDisappearing

cleaned up some comments

yougotwill committed:
refactor: update region comments to #region

yougotwill committed:
refactor: rename expirationLength

to expirationDurationMs

yougotwill committed:
fix: more comment cleanup

yougotwill committed:
fix: remove unneeded export for

writeSessionSchemaVersion

yougotwill committed:
fix: more comment cleanup

yougotwill committed:
fix: remove unneeded Boolean cast

yougotwill committed:
fix: move isDelete into inbox poll if statement

fixed some comments

yougotwill committed:
refactor: ttl_config to config_message

inside of TTL_DEFAULT. This is more readable and explicit

yougotwill committed:
refactor: existingexpration to existingexpirationt

in createClosedGroups.ts. Cleaned up some comments

yougotwill committed:
fix: closed group groupId is now PubKey only

yougotwill committed:
fix: foundMessage has the ttl updated by reference

no need to explicitly assign it

yougotwill committed:
feat: added checkForExpiringOutgoingMessage

triggers timer for all outgoing messages. includes unit tests

yougotwill committed:
feat: added todo to section duck

we want to move it into a redux slice in future

yougotwill committed:
fix: use switch instead of chained ifs

in setExpirationStartTimestamp

yougotwill committed:
refactor: move checkShouldDisappearButIsntMessage

to legacy file

yougotwill committed:
fix: throw error in getMessageReadyToDisappear

for community messages

yougotwill committed:
fix: comment cleanup

yougotwill committed:
refactor: rename ttl_max to content_message

this is more explicit

yougotwill commented on ts/state/selectors/user.ts:
```diff
@@ -12,4 +12,9 @@ export const getOurNumber = createSelector(
   (state: UserStateType): string => state.ourNumber
 );
 
+export const getOurDisplayNameInProfile = createSelector(
+  getUser,
+  (state: UserStateType): string => state.ourDisplayNameInProfile
+);
+
```
Keeping until we remove the legacy code.

yougotwill committed:
chore: made note about the hasOutdatedClient prop

we will remove it when we remove the legacy code

yougotwill commented on ts/state/smart/SessionConversation.ts:
```diff
@@ -28,6 +28,7 @@ const mapStateToProps = (state: StateType, ownProps: SmartSessionConversationOwn
     selectedConversationKey: getSelectedConversationKey(state),
     theme: getTheme(state),
     messagesProps: getSortedMessagesOfSelectedConversation(state),
+    ourDisplayNameInProfile: getOurDisplayNameInProfile(state),
```
Chatted about it and keeping until we remove the legacy code

yougotwill commented on ts/components/conversation/right-panel/overlay/disappearing-messages/OverlayDisappearingMessages.tsx:
```diff
@@ -0,0 +1,260 @@
+  weAreAdmin: boolean | undefined;
+};
+
+export const OverlayDisappearingMessages = () => {
```
Fixed by creating custom hooks for the legacy useEffects

yougotwill committed:
feat: extra legacy useEffects into custom hooks

for OverlayDisappearingMessages

yougotwill committed:
fix: remove unneeded handleSetTime

in OverlayDisappearingMessages

yougotwill commented on ts/components/conversation/right-panel/overlay/disappearing-messages/OverlayDisappearingMessages.tsx:
```diff
@@ -0,0 +1,260 @@
+    }
+  };
+
+  const handleSetTime = (value: number) => {
+    setTimeSelected(value);
+  };
```
Fixed

yougotwill commented on ts/components/conversation/right-panel/overlay/disappearing-messages/OverlayDisappearingMessages.tsx:
```diff
@@ -0,0 +1,260 @@
+    );
+  }, [expireTimer, hasOnlyOneMode, modeSelected]);
+
+  // useEffect(() => {
+  //   window.log.debug(
+  //     `WIP: OverlayDisappearingMessages: \nmodeSelected ${modeSelected} \nsingleMode ${singleMode} \ntimeSelected ${timeSelected} \ntimerOptions ${JSON.stringify(
+  //       timerOptions.map(option => option.name)
+  //     )}`
+  //   );
+  // }, [modeSelected, timeSelected]);
```
Fixed

yougotwill commented on ts/state/selectors/conversations.ts:
```diff
@@ -871,6 +872,29 @@ export const getMessageAttachmentProps = createSelector(getMessagePropsByMessage
   return msgProps;
 });
 
+export const getMessageExpirationProps = createSelector(getMessagePropsByMessageId, (props):
+  | PropsForExpiringMessage
+  | undefined => {
+  if (!props || isEmpty(props)) {
+    return undefined;
+  }
+
+  const msgProps: PropsForExpiringMessage = {
+    ...pick(props.propsForMessage, [
+      'convoId',
+      'direction',
+      'receivedAt',
+      'isUnread',
+      'expirationTimestamp',
+      'expirationLength',
+      'isExpired',
+    ]),
+    messageId: props.propsForMessage.id,
+  };
+
+  return msgProps;
+});
+
```
Fixed

yougotwill committed:
refactor: moved getMessageExpirationProps

into useParamSelector since it is only called there

Bilb committed:
Merge pull request #2956 from KeeJef/readme-update

docs: update I18n section

yougotwill committed:
feat: cleaned up wip logs

Bilb commented on ts/util/releaseFeature.ts:
```diff
@@ -118,12 +119,11 @@ export const ReleasedFeatures = {
 
-// DO NOT MERGE
-// TODO Remove before PR
+// TODO DO NOT MERGE Remove after QA
```
we need to remember this

yougotwill replied:
I've made a note in my journal for this.

Bilb commented on ts/window.d.ts:
```diff
@@ -105,8 +105,7 @@ declare global {
     updateZoomFactor: () => void;
-    // DO NOT MERGE
-    // TODO Remove before PR
+    // TODO DO NOT MERGE Remove after QA
```
same

yougotwill replied:
👍

Bilb commented on ts/test/session/unit/disappearing_messages/DisappearingMessage_test.ts:
```diff
@@ -0,0 +1,679 @@
+    it("if it's a Private Conversation and the expirationMode is deleteAfterRead and expireTimer > 0 then the message's expirationType is deleteAfterRead", async () => {
+      const conversation = new ConversationModel({
+        ...conversationArgs,
+      } as any);
```
we could name those as ConversationAttributes. Just because we want to  have the least amount of any in the app (and it would be nice to not have those when searching for : any)

yougotwill commented on ts/components/conversation/SessionMessagesListContainer.tsx:
```diff
@@ -51,16 +51,22 @@ type Props = SessionMessageListProps & {
+  padding-top: var(--margins-sm);
+  padding-right: var(--margins-lg);
+  padding-bottom: var(--margins-xl);
+  padding-left: ${props =>
+    props.isGroup ? 'calc(var(--margins-lg) + 11px)' : 'var(--margins-lg)'};
```
You are a wizard bilby 🪄

yougotwill committed:
fix: explicitly set isCentered

yougotwill committed:
refactor: export functions from an object for stubbing

for disappearing messages index file

yougotwill committed:
fix: comment typos

yougotwill committed:
test: replace any with ConversationAttributes

on convo constructor args

yougotwill committed:
fix: comment note casing

yougotwill committed:
fix: no any in useSelectors

for getMessageAttachmentProps and getIsMessageSelected

Bilb commented on ts/components/basic/SessionRadio.tsx:
```diff
@@ -150,6 +166,7 @@ export const SessionRadioPrimaryColors = (props: {
         outlineOffset={outlineOffset}
+        disabled={disabled}
       >
         {''}
```
I suppose I was taking about the {'"} just below the line of my comment

Bilb commented on ts/hooks/useParamSelector.ts:
```diff
@@ -264,6 +267,78 @@ export function useIsTyping(conversationId?: string): boolean {
+            option.value === 86400 || // 1 day
+            option.value === 604800 || // 1 week
+            option.value === 1209600 // 2 weeks
+          );
```
works for me

Bilb commented on ts/session/apis/snode_api/getExpiriesRequest.ts:
```diff
@@ -0,0 +1,238 @@
+  const ourPubKey = UserUtils.getOurPubKeyStrFromCache();
+  if (!ourPubKey) {
+    window.log.error('WIP: [getExpiriesFromSnode] No pubkey found', messageHashes);
+    return [];
```
it's just that if the user does not have a pubkey, there is a very big issue. And there is probably no point of doing anything

Bilb commented on ts/components/conversation/ExpireTimer.tsx:
```diff
@@ -35,8 +15,14 @@ const ExpireTimerBucket = styled.div`
 `;
 
+type Props = {
+  expirationLength: number; // should be in milliseconds
```
yep

Bilb commented on ts/components/conversation/header/ConversationHeaderSubtitle.tsx:
```diff
@@ -0,0 +1,163 @@
+
+      return (
+        <StyledSubtitleDot
+          key={`subtitleDotMenu-${index}`}
```
do we need the index though?
having the index is the bad practice!

Bilb commented on ts/session/utils/calling/CallManager.ts:
```diff
@@ -814,8 +840,15 @@ export async function USER_acceptIncomingCallRequest(fromSender: string) {
   }
-  window.inboxStore?.dispatch(answerCall({ pubkey: fromSender }));
-  await openConversationWithMessages({ conversationKey: fromSender, messageId: null });
+  window.inboxStore?.dispatch(
+    answerCall({
+      pubkey: fromSender,
+    })
+  );
+  await openConversationWithMessages({
+    conversationKey: fromSender,
+    messageId: null,
+  });
```
same

Bilb commented on ts/session/apis/snode_api/getExpiriesRequest.ts:
```diff
@@ -0,0 +1,238 @@
+      expireRequest.params.messages
+    );
+
+    if (!Object.keys(expirationResults).length) {
+      throw new Error('expirationResults is empty');
+    }
+
+    const expiryTimestamps: Array<number> = Object.values(expirationResults);
```
Map and Set? But that's not something to fix for now.

yougotwill replied:
👍

Bilb commented on ts/util/readReceipts.ts:
```diff
@@ -44,26 +44,20 @@ async function onReadReceipt(receipt: { source: string; timestamp: number; readA
-      await message.commit();
-    }
+    await message.commit();
 
```
up for this?

yougotwill committed:
fix: alignment for group messages with no avatar

this happens with multiple incoming messages

yougotwill committed:
fix: merge now returns good hashes

we calc the latestEnvelopeTimestamp and use for disappearAfterRead

yougotwill committed:
chore: bump libsession_util_nodejs to v0.3.0

yougotwill committed:
Merge branch 'unstable' into userconfig_disappearingmessage

yougotwill commented:
@burtonemily the time has come 🫡

yougotwill committed:
fix: use correct values for legacy mode

fix datatestid on generic message

Bilb committed:
chore: disappear feature switch based on env variable for QA

Bilb committed:
fix: move right panel buttons to a panelbuttongroup

also enforce a 300 min width for the right panel, otherwise the bloaded
disappearing message design is not looking good

Bilb committed:
fix: outgoing start call notification timer start on after read

Bilb committed:
chore: add missing accessibilityIds for menu buttons/radio

Bilb committed:
fix: send extraction notification even if read receipts is off

Bilb committed:
Revert "fix: move right panel buttons to a panelbuttongroup"

This reverts commit e6dd6844f966650f7c102c5c538fbfbda8550b30.

Bilb committed:
feat: use multi value to know if disappear_v2 is live or nah

yougotwill committed:
Merge pull request #37 from Bilb/userconfig_disappearingmessage

QA fixes

yougotwill committed:
fix: off control msgs expire on the last setting

Bilb committed:
chore: datatestid cannot start with a digit (I told you so)

Bilb committed:
fix: datatestIds cannot have spaces

Bilb committed:
fix: force text to black for noticebanner light theme

Bilb committed:
fix: allow answered&missed call notification to disappear after read

Bilb committed:
fix: handle dataextraction notification even if read receipts are off

Bilb committed:
fix: send sync message with the expireUpdate if provided

Bilb committed:
feat: merge expiring stopwatch and messagestatus together

Bilb committed:
fix: hide known message status except for last message

Bilb committed:
fix: expiration update to off expire with previous expiration settings

Bilb committed:
fix: update redux mostRecentMessageid anytime a message changes/added

Bilb committed:
fix: start timer on message even if message already exists

Bilb committed:
fix: fix typo pushing new messages to redux store

Bilb committed:
fix: made the saved media notification a bubble

as it is not pretty otherwise

Bilb committed:
fix: avatar style for incoming messages in groups

had to refactor a fair bit, but well...

Bilb committed:
chore: remove dtrace-provider dep from bunyan using our own

Bilb committed:
fix: fix macos CI by forcing use of python 3.11

Bilb committed:
fix: glitch of background with crownicon and css translate

Bilb committed:
fix: legacy group invite do not expire

Bilb committed:
feat: filter out sub 60s options for expiretimer when packaged app

Bilb committed:
fix: do not auto switch to non-legacy on overlay opening

when v2 is released we don't want to auto switch the overlay right panel
anymore

Bilb committed:
fix: address PR reviews

Bilb committed:
fix: broken state adding lastDisappearingMessage to chatmsg

Bilb committed:
chore: remove lastChangetimestmap and add explciit expiretype and type

on the messages creation

Bilb committed:
fix: drop support for argon2 based ons

Bilb committed:
fix: add most of new designs (own setting) for disappear v2 messages

Bilb committed:
feat: add cleanup of old expire update in history

we only keep one from each sender

Bilb committed:
fix: show date break if we don't have a previous message

Bilb committed:
fix: already read DaR messages start with right time left on receive

Bilb committed:
Merge pull request #2994 from oxen-io/unstable

Session 1.11.5

Bilb committed:
fix: add jobs for expiry update & expiry fetch

Bilb committed:
fix: use expiry from swarm to update readAt & expiresAt for msg

Bilb committed:
fix: use DaR ttl for synced message store request

removing one request at a time

Bilb committed:
chore: incoming group update message need their own expireTimer

we do not trust the setting from the convo anymore

Bilb committed:
fix: bug with updatelastmessage not resetting when no msgs

also make fix the DaS expires_at so destroyExpiredMessages finds it

Bilb committed:
fix: changed some message control to not expire

still some to discuss with the team

Bilb committed:
fix: use at least 2 hashes for the update_expiries too

until the storage server release is live we need this workaround

Bilb committed:
test: fix unit tests

Bilb committed:
fix: make group control message not expire

Bilb committed:
fix: call messages expires with recipients expire settings

Bilb committed:
fix: make DataExtractionNotification msg stay in swarm for 14d

Bilb committed:
fix: make callMessage forced to be DaR for recipient

and put workaround for our local message

Bilb committed:
fix: do not show DaR/S followsetting button for legacy

Bilb committed:
Merge remote-tracking branch 'oxen/unstable' into userconfig_disappearingmessage

Bilb committed:
fix: make fileserver request retry

Bilb committed:
chore: wrap text of dialog desc when more than 90c long

Bilb committed:
fix: typo for isControlMessage()

Bilb committed:
chore: to remove. force disappear v2 ON temporarly

Bilb committed:
fix: das group control message go not expire

Bilb committed:
chore: bump to session 1.11.6

Bilb committed:
Merge remote-tracking branch 'oxen/clearnet' into userconfig_disappearingmessage

Bilb committed:
chore: lint

Bilb committed:
fix: fix crash on password window opening

Bilb commented:
closing this in favor of https://github.com/oxen-io/session-desktop/pull/3016
